src/main/java/com/softserve/academy/spaced/repetition/controller/CourseCommentController.java

```diff
@@ -34,70 +34,47 @@
     private CourseCommentService courseCommentService;
 
     @Auditable(action = AuditingAction.VIEW_ALL_COMMENTS_FOR_COURSE)
-    @GetMapping(value = "/api/category/{categoryId}/courses/{courseId}/comments")
-    public ResponseEntity<List<CommentDTO>> getAllCommentsByCourse(@PathVariable Long categoryId,
-                                                                   @PathVariable Long courseId) {
+    @GetMapping
+    public ResponseEntity<List<CommentDTO>> getAllCommentsByCourseId(@PathVariable Long courseId) {
         LOGGER.debug("View all comments for course with id: {}", courseId);
         List<Comment> courseCommentsList = courseCommentService.getAllCommentsForCourse(courseId);
         Link collectionLink = linkTo(methodOn(CourseCommentController.class)
-                .getAllCommentsByCourse(categoryId, courseId)).withSelfRel();
+                .getAllCommentsByCourseId(courseId)).withSelfRel();
```

Comment:
Maybe `getCommentByCourseId`, because you return list of comments and each comment has a link to self, and this link we can get from method `getCommentByCourseId`.

Reply:
it`s no necessary